classes/CartRule.php

```diff
@@ -1328,12 +1328,7 @@ public function getContextualValue($use_tax, Context $context = null, $filter =
                             && (($this->reduction_exclude_special && !$product['reduction_applies']) || !$this->reduction_exclude_special)) {
                             $price = $product['price'];
                             if ($use_tax) {
-                                $infos = Product::getTaxesInformations($product, $context);
-                                $tax_rate = $infos['rate'] / 100;
-                                // As the price is tax excluded but ecotax included, we need to substract the ecotax before getting the price tax included
-                                $price -= $product['ecotax'];
-                                $price *= (1 + $tax_rate);
-                                $price += $product['ecotax'];
+                                $price = $product['price_without_reduction'];
```

Comment:
Hi @mflasquin @jolelievre @matks,
A bit late but why use `$product['price_without_reduction']` here ?

`$product['price']` is the current price without tax and with reduction apply.
So, in this case, we want to use the current price with tax AND with reduction apply (ie `$product['price_with_reduction']` variable)
With the use of `$product['price_without_reduction']`, it means that we use the current price with tax but without reduction, so it's ok for all product without reduction but if we hare a product with a specific price apply, the cart rule amount is wrongly computed